src/github.com/getlantern/balancer/balancer.go

```diff
@@ -22,21 +22,30 @@ var (
 // Balancer balances connections established by one or more Dialers.
 type Balancer struct {
 	dialers []*dialer
+	trusted []*dialer
 }
 
 // New creates a new Balancer using the supplied Dialers.
 func New(dialers ...*Dialer) *Balancer {
-	dhs := make([]*dialer, 0, len(dialers))
+	bal := new(Balancer)
+
+	// Sort dialers by QOS (ascending) for later selection
+	sort.Sort(byQOSAscending(dialers))
```

Comment:
Now that you mention it, yes, this practice could lead to unexpected behaviour in any piece of code that depends on the variable that was passed. I'd agree with the change.